src/elf/sym.rs

```diff
@@ -94,9 +94,9 @@ pub fn st_visibility(other: u8) -> u8 {
 
 /// Is this information defining an import?
 #[inline]
-pub fn is_import(info: u8, value: u64) -> bool {
+pub fn is_import(info: u8, shndx: u64) -> bool {
     let bind = st_bind(info);
-    bind == STB_GLOBAL && value == 0
+    (bind == STB_GLOBAL || bind == STB_WEAK) && shndx == SHN_UNDEF as _
```

Comment:
this is a good catch (the missing STB_WEAK), but changing this function to operate semantically on `shndx` values instead of `st_value`, while backwards compatible, has wider implications.

for example, any clients relying on `st_value` being used in these computations will now break when section headers are stripped.

imho my general thesis that what a dynamic linker can import is how we've defined import, while arguably more nuanced than this, is a good one and has value (no pun intended). using `st_value` is a heuristic for that thesis.

that being said, it seems like people are hitting some issues with this, so we can consider adding something like `is_import_from_sections` or something which is new api to opt into using the section header (perhaps along with st_value) as the determinant.